src/codeu/chat/server/Transaction.java

```diff
@@ -0,0 +1,161 @@
+package codeu.chat.server;
+
+import codeu.chat.common.User;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.Tokenizer;
+import codeu.chat.util.Time;
+import codeu.chat.server.Controller;
+
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import java.io.FileWriter;
+import java.io.File;
+
+import java.io.IOException;
+
+
+public final class Transaction {
+
+  private ArrayList<String> transactionsList = new ArrayList<String>();
+  private Controller controller;
+
+  public Transaction(Controller controller) {
+    this.controller = controller;
+    read();
+  }
+
+  // takes in information that the server needs to populate itself!
+  private void read() {
+    File transaction = new File("transaction.txt");
+    try {
+      Scanner in = new Scanner(transaction);
+      while (in.hasNextLine()) {
+        String line = in.nextLine();
+        System.out.println(line);
+        Tokenizer tokens = new Tokenizer(line);
+        tokens.next();
+        switch (tokens.next()) {
+          case "CREATEUSER" : readUser(tokens); break;
+          case "CREATEMESSAGE" : readMessage(tokens); break;
+          case "CREATECONVO" : readConvo(tokens); break;
+        }
+      }
+    }
+    catch (IOException e) {
+    }
+  }
+
+  private void readUser(Tokenizer tokens) {
+    try {
+      tokens.next();
+      Uuid id = Uuid.parse(tokens.next());
+      tokens.next();
+      String name = tokens.next();
+      tokens.next();
+      Time creation = Time.fromMs(new Long(tokens.next()));
+      controller.newUser(id, name, creation);
+    }
+    catch (IOException e) {
```

Comment:
Similar to what I said before, there are very few reasons exceptions should ever be swallowed.  I don't think this is a good reason to swallow exceptions here or in any of the read commands.